std/uuid.d

```diff
@@ -1058,18 +1058,21 @@ unittest
  *      randomGen = uniform RNG
  * See_Also: $(XREF random, isUniformRNG)
  */
-@trusted UUID randomUUID(RNG)(ref RNG randomGen) if(isIntegral!(typeof(RNG.front)))
+UUID randomUUID(RNG)(ref RNG randomGen) if(isIntegral!(typeof(RNG.front)))
 {
     import std.random : isUniformRNG;
     static assert(isUniformRNG!RNG, "randomGen must be an uniform RNG");
-    enum size_t elemSize = typeof(RNG.front).sizeof;
+
+    alias E = ElementEncodingType!RNG;
+    enum size_t elemSize = E.sizeof;
     static assert(elemSize <= 16);
+    static assert(16 % elemSize == 0);
+
     UUID u;
-    for(size_t i; i < 16; i += elemSize)
+    for (size_t i = 0 ; i < 16 ; i += elemSize)
     {
         randomGen.popFront();
-        immutable randomValue = randomGen.front;
-        u.data[i .. i + elemSize] = (*cast(ubyte[elemSize]*)&randomValue)[];
+        unsafeTrustingArrayElementReinterpret!E(u.data, i) = randomGen.front;
```

Comment:
Yeah, this code looks broken. First of all, the sig constraint fails to check whether `RNG` is actually a valid RNG -- it will happily instantiate a non-range type that just happens to have an integral member named `front`. Secondly, calling `.front` after `popFront` without checking for `empty` is technically UB according to the range API. Of course, this wouldn't be a problem if the sig constraint checking for the RNG also ensures that RNGs are infinite ranges (they should be, as otherwise I can't conceive of any meaningful use for them!). But the `popFront` call means it will discard the first random value from the RNG for no particular reason, *and* leave the *last* value from the RNG still in `.front` for the next user of the RNG to pick up. Not good.